### PR TITLE
Peekasync

### PR DESCRIPTION
**Pull request by nemakam**
## Description
Introduced PeekAsync() functionality for MessageReceivers.

Three new public APIs have been added:
1. Task<BrokeredMessage> PeekAsync() 
    --> This peeks one message.
2. Task<IList<BrokeredMessage>> PeekAsync(int maxMessageCount)
    --> This peeks a maximum of 'maxMessageCount' messages.
3. Task<BrokeredMessage> PeekBySequenceNumberAsync(long fromSequenceNumber)
    --> This peeks a message with a seq number greater than 'fromSequenceNumber'

Added test cases for the same.

### CODE REVIEW

File: src/Microsoft.Azure.ServiceBus/MessageReceiver.cs [hunk 1]
```diff
@@ -111,9 +131,20 @@ public Task<DateTime> RenewLockAsync(Guid lockToken)
             return this.OnRenewLockAsync(lockToken);
         }
 
-        internal Task<AmqpResponseMessage> ExecuteRequestResponseAsync(AmqpRequestMessage amqpRequestMessage)
+        public Task<BrokeredMessage> PeekAsync()
```
Comment (jtaubensee) [inline src/Microsoft.Azure.ServiceBus/MessageReceiver.cs]: @nemakam - Can we add the proper documentation for these new public methods?

Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/MessageReceiver.cs]: @jtaubensee , I agree we need to have documentation for all the public methods (new and old). I will send a PR later with the documentation. Its better to get the whole thing written together for uniformity.

Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/MessageReceiver.cs]: @jtaubensee , I've added the documentation for the new public methods :)

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpResponseMessage.cs [hunk 2]
```diff
@@ -79,5 +79,12 @@ public IEnumerable<TValue> GetListValue<TValue>(MapKey key)
 
             return list.Cast<TValue>();
         }
+
+        public AmqpSymbol GetResponseErrorCondition()
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpResponseMessage.cs]: I can't seem to find where this method is used in PR. At the same time, [`AmqpExceptionHelper`](https://github.com/Azure/azure-service-bus-dotnet/blob/peekasync/src/Microsoft.Azure.ServiceBus/Amqp/AmqpExceptionHelper.cs#L39) has it as well.

File: src/Microsoft.Azure.ServiceBus/MessageReceiver.cs [hunk 3]
```diff
@@ -43,6 +45,24 @@ public virtual int PrefetchCount
...
+            get
+            {
+                return this.lastPeekedSequenceNumber;
+            }
+
+            internal set
+            {
+                if (value < 0)
+                {
+                    throw new ArgumentOutOfRangeException("LastPeekedSequenceNumber", value.ToString());
```
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/MessageReceiver.cs]: > ArgumentOutOfRangeException("LastPeekedSequenceNumber", value.ToString());

ArgumentOutOfRangeException(nameof(LastPeekedSequenceNumber), value.ToString());

**Comment by jtaubensee** [comment]
Merged recent changes to incorporate test changes that will create session specific topics

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpResponseMessage.cs (hunk 2 above)
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpResponseMessage.cs]: My bad. I removed another function which was referencing this and didn't clean this up.

File: src/Microsoft.Azure.ServiceBus/MessageReceiver.cs (hunk 3 above)
Comment (nemakam) [inline src/Microsoft.Azure.ServiceBus/MessageReceiver.cs]: Resolved

File: src/Microsoft.Azure.ServiceBus/Amqp/AmqpResponseMessage.cs (hunk 2 above)
Comment (SeanFeldman) [inline src/Microsoft.Azure.ServiceBus/Amqp/AmqpResponseMessage.cs]: No worries. I was wondering if I'm missing something 😃

File: src/Microsoft.Azure.ServiceBus/MessageReceiver.cs (hunk 3 above)
Comment (jtaubensee) [inline src/Microsoft.Azure.ServiceBus/MessageReceiver.cs]: @nemakam - Did you push your changes for this?

**Pull request merged**